### PR TITLE
ki.SaveJSON does not work.

### PR DESCRIPTION
**Pull request by massn**
Hi! I'm Massn.

I used your great library, but `ki.SaveJson` did not work. This function makes a file, but writes nothing.
I found a solution for it.

Thanks!

### CODE REVIEW

File: ki/node_test.go [hunk 1]
```diff
@@ -862,7 +862,7 @@ func TestNodeFieldFunc(t *testing.T) {
 	res = res[:0]
 }
 
-func TestNodeFieldJSonSave(t *testing.T) {
+func TestNodeFieldJSONSave(t *testing.T) {
```
Comment (massn) [inline ki/node_test.go:865]: typo?

File: ki/node_test.go [hunk 2]
```diff
@@ -379,7 +379,7 @@ func TestNodeConfig(t *testing.T) {
 //////////////////////////////////////////
 //  JSON I/O
 
-func TestNodeJSonSave(t *testing.T) {
+func TestNodeJSONSave(t *testing.T) {
```
Comment (massn) [inline ki/node_test.go:382]: typo?

File: ki/io.go [hunk 3]
```diff
@@ -62,12 +62,12 @@ func (n *Node) WriteJSON(writer io.Writer, indent bool) error {
...
 	copy(nwb, []byte(tstr))
 	copy(nwb[len(tstr):], b) // is there a way to avoid this?
+
 	_, err = writer.Write(nwb)
 	if err != nil {
 		log.Println(err)
-		return err
 	}
-	return nil
+	return err
```
Comment (massn) [inline ki/io.go:70]: refactored

**Comment by rcoreilly** [comment]
Thanks for the fixes!  I just pushed a fix that keeps the bufio.Writer which is important for speed.  I didn't realize it doesn't auto-flush when the file closes so I added a flush after writing (hadn't been using buffered writes before, but saw that it is very important for speed).  Give it a try.

**Comment by massn** [comment]
Thanks for your quick response!
I tried your new version (`v1.1.1`) and it works well.

Thank you very much!

**Pull request closed without merging**